sdk/monitor/monitor-ingestion/src/utils/concurrentPoolHelper.ts

```diff
@@ -1,19 +1,26 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 
+import { AbortSignalLike } from "@azure/abort-controller";
+
 export async function concurrentRun<T>(
   maxConcurrency: number,
   inputData: Array<T>,
-  callback: (args: T) => Promise<void>
+  callback: (args: T) => Promise<void>,
+  abortSignal?: AbortSignalLike
 ): Promise<void> {
   const dataQueue = [...inputData].reverse();
   const promises: Array<Promise<void>> = [];
 
-  function removePromise(p: Promise<void>) {
+  function removePromise(p: Promise<void>): void {
     promises.splice(promises.indexOf(p), 1);
   }
   while (dataQueue.length) {
     while (dataQueue.length && promises.length < maxConcurrency) {
+      if (abortSignal?.aborted) {
+        await Promise.all(promises);
```

Comment:
I think it is worth adding a comment saying that here, we'll stop processing new calls and just wait for the existing ones to resolve/reject

Reply:
might also be good to put a logger.info

Reply:
We decided to remove that code, since it's never hit